leaguecog/leaguecog.py

```diff
@@ -139,15 +147,156 @@ async def league(self, ctx: commands.Context):
+            number_emojis = ReactionPredicate.NUMBER_EMOJIS[: len(text_channel_dict)]
+
+            msg += "React with the appropriate channel number:\n\n"
+
+            # number the channels within the embed message with the emojis
+            for emoji, channel in zip(number_emojis, text_channel_dict.keys()):
+                msg += f"{emoji} {channel}\n"
+
+            channel_embed = await Ezreal.build_embed(
+                self,
+                title="SETUP - ANNOUNCEMENT CHANNEL",
+                msg=msg,
+            )
+            channel_msg = await ctx.send(embed=channel_embed)
+
+            # set up a ReactionPredicate and index text_channel_dict based on response
+            start_adding_reactions(channel_msg, number_emojis)
+            channel_pred = ReactionPredicate.with_emojis(number_emojis, channel_msg, ctx.author)
+            await ctx.bot.wait_for("reaction_add", check=channel_pred)
+
+        else:
+            # add code blocking for looks
+            msg += "Input the appropriate channel number:\n```"
+            for idx, ch in enumerate(text_channel_dict.keys()):
+                msg += f"{idx}. {ch}\n"
+            msg += "```"
+            channel_embed = await Ezreal.build_embed(
+                self,
+                title="SETUP - ANNOUNCEMENT CHANNEL",
+                msg=msg,
+            )
+            channel_msg = await ctx.send(embed=channel_embed)
+            # make sure the response is an integer that corresponds to a channel
+            #   but MessagePredicate will read these as strings, not ints
+            channel_pred = MessagePredicate.contained_in(
+                [str(i) for i in range(len(text_channel_dict))]
+            )
+            await ctx.bot.wait_for("message", check=channel_pred)
+
+        # index the channel name and id from the dict
+        channel_idx = channel_pred.result
+        alert_channel_name = [k for k in text_channel_dict.keys()][channel_idx]
+        alert_channel_id = text_channel_dict[alert_channel_name]
+
+        log.info(f"SETUP channel_pred.result == {channel_pred.result}")
+        log.info(f"SETUP alert_channel_name = {alert_channel_name}")
+        log.info(f"SETUP alert_channel_id == {alert_channel_id}")
+
+        # set the alert channel via channel id
+        await self.config.guild(ctx.guild).alertChannel.set(alert_channel_id)
+        log.info(
+            f"SETUP self.config.guild(ctx.guild).alertChannel() == {await self.config.guild(ctx.guild).alertChannel()}"
+        )
+
+        # TODO remove reactions, or remove last message if had to get text input
+
+        # edit the original embed and show the user which channel was selected
+        channel_embed = await Ezreal.build_embed(
+            self,
+            title="SETUP - ANNOUNCEMENT CHANNEL",
+            msg=f"Annoucement channel set to #{alert_channel_name}",
+        )
+        await channel_msg.edit(content=ctx.author.mention, embed=channel_embed)
+
```

Comment:
I also would handle a timeout scenario. Maybe you did and I didn't catch it. What if the user never responds to a message?
Something is weird here too, lots of session errors when ending the bot that were not here before. I assume each predicate is some kind of session?